cfgov/v1/handlers/blocks/feedback.py

```diff
@@ -43,14 +41,29 @@ def __init__(self,
         super(FeedbackHandler, self).__init__(page, request)
         self.block_value = block_value
 
-    def process(self, is_submitted):
+    def sanitize_referrer(self):
+        known_miscodings = {
+            u'\xc3\xb3': u'\xf3',
+            u'\xc3\xa9': u'\xe9',
+            u'\xc3\xad': u'\xed',
+        }
+        referrer = self.request.META.get('HTTP_REFERER', '')
+        try:
+            referrer.encode('ascii')
+        except (UnicodeEncodeError):
+            for char in known_miscodings:
+                referrer = referrer.replace(char, known_miscodings[char])
+            referrer = urllib.quote(referrer.encode('utf8'), safe=':/')
+        return referrer
 
+    def process(self, is_submitted):
         form_cls = FEEDBACK_TYPES[get_feedback_type(self.block_value)]
 
         if is_submitted:
             form = form_cls(self.request.POST)
             return self.get_response(form)
 
+        self.block_value.update({'referrer': self.sanitize_referrer()})
```

Comment:
I worry a bit about overloading the block value like this, as it depends on the inner workings of the Wagtail block rendering. Am I reading this right that you're mocking out a `feedback` attribute on the block [even though one doesn't exist](https://github.com/cfpb/cfgov-refresh/blob/ca9053d66e65ab0f66490fab6f08a4b7354354f8/cfgov/v1/blocks.py#L75)?

Is there some other way of passing this to the context, e.g. add it to the 

```py
return {
    'form': form_cls(),
    'referrer': self.sanitize_referrer(),
}
```

and then in your template refer to `settings.referrer`?

Reply:
Thank you for this suggestion.